src/Authenticator/ResultInterface.php

```diff
@@ -17,32 +17,32 @@
     /**
      * General Failure
      */
-    const FAILURE = 0;
+    const FAILURE = 'FAILURE';
 
     /**
      * Failure due to identity not being found.
      */
-    const FAILURE_IDENTITY_NOT_FOUND = -1;
+    const FAILURE_IDENTITY_NOT_FOUND = 'FAILURE_IDENTITY_NOT_FOUND';
 
     /**
-     * Failure due to invalid credential being supplied.
+     * Failure due to invalid credentials being supplied.
      */
-    const FAILURE_CREDENTIAL_INVALID = -2;
+    const FAILURE_CREDENTIALS_INVALID = 'FAILURE_CREDENTIALS_INVALID';
 
     /**
-     * Failure due to other circumstances.
+     * The authentication credentials were not found in the request.
      */
-    const FAILURE_OTHER = -3;
+    const FAILURE_CREDENTIALS_NOT_FOUND_IN_REQUEST = 'FAILURE_CREDENTIALS_NOT_FOUND_IN_REQUEST';
```

Comment:
This constant name is too long, `_IN_REQUEST` should be dropped.

Reply:
would `FAILURE_CREDENTIALS_NOT_IN_REQUEST` instead of `FAILURE_CREDENTIALS_NOT_FOUND` be ok?

Reply:
I think the `_IN_REQUEST` bit is the frivolous bit. `FAILURE_CREDENTIALS_NOT_FOUND` does the trick, as @ADmad suggested,

Reply:
@ADmad I changed this again... are you okay with this, now?